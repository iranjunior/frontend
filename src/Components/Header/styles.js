import styled from 'styled-components';
import { makeStyles } from '@material-ui/core/styles';
export const useStyles = makeStyles({
    avatar: {
      height: 30,
      width: 30,
    },
    menu: {
      top: 40
    },
    icon_button: {
      height: 30,
      width: 30,
    },
    bigAvatar: {
      margin: 10,
      width: 60,
      height: 60,
    },
    iconAvatarList:{
        marginRight: 6
    }
  });
  
export const Headers = styled.header`
    
        width: 100%;
        background: linear-gradient(45deg, #0288d1 0%, #26c6da 100%);
        position: absolute;
        top: 0;
        left: 0;
        z-index: -1000;
        box-shadow: 1px 1px 2px 1px rgba(10,10,10,0.3);
    
`;
export const List = styled.ul`
    list-style-type: none;
    display: flex;
    flex-flow: row;
    flex-direction: row;
    justify-content: space-between;
    margin: 2px;
`;
export const Element = styled.li`
    @media only screen and (min-width: 700px){
        display: block;
        margin: auto;
        padding: auto;
        :nth-child(1){
            display: none;
        }
        :nth-child(3){
            width: 70%
        }
    }
    @media only screen and (max-width: 700px){
        display: none;
        :nth-child(1){
            display: block;
        }
    }
`;
export const Text = styled.input.attrs({type: 'text'})`
    width: 100%;
    height: 100%;
    background: rgba(255,255, 255, 0.3);
    border: none;
    border-radius: 3px;
    padding: 10px;
    ::placeholder{
        color: #fff;
    }
    &:hover{
        background: rgba(255,255, 255, 0.5);

    }  
`;
export const FieldText = styled.div`
    margin: auto;
    padding: auto;
    width: 80%;
`;
export const View = styled.div`
    padding: 7px;
`;

export const Circle = styled.div`
    position: relative;
    z-index: -10;
    background: #ff4081;
    height: 40px;
    width: 40px;
    border-radius: 50%;
`;
export const IconSpace = styled.div`
    margin-right: 10px;
    margin-top: 5px;
`;
