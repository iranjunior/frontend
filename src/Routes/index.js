import React from 'react';
import { connect } from 'react-redux';
import { Switch, Route } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router'
import PrivateRoute   from './privateRoute';
import history from './history'

import Home  from '../Pages/Home';
import Login  from '../Pages/Login';
//import isAuthenticate  from '../Services/auth';

//console.log(isAuthenticate(2));


const Routes = (  ) => {
    return (
        <ConnectedRouter history={history}>
            <Switch>
                <PrivateRoute path='/' exact component={Home}/>
                <Route path='/login'  component={Login}/>
            </Switch>
        </ConnectedRouter>
    )
}
export default connect(state => ({ }))(Routes);