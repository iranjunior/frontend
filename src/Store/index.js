import { createStore } from "redux"

const INITIAL_STATE = {
    token: null,
    user: {},
    informations: {}
}

function reducers(state = INITIAL_STATE, action) {
    console.log(state);
    switch (action.type) {
        case 'EMAIL_USER':
            return{
                ...state,
                user:{ ...state.user, email: action.email}
            }        
        case 'PASSWORD_USER':
            return{
                ...state,
                user:{ ...state.user,  password: action.password}
            }
        case 'LOGGED_USER':
            return{
                ...state,
                token: action.token
            }
        default:
            return state;
}
}

const store = createStore(reducers)

export default store;