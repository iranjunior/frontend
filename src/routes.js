import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home  from './Pages/Home';
import Login  from './Pages/Login';
const Routes = () => {
    return (
        <Switch>
            <Route path='/' exact component={Home}/>
            <Route path='/login' exact component={Login}/>
        </Switch>
    )
}
export default Routes;