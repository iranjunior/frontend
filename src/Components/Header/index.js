import React from 'react';
import { IconContext } from 'react-icons'
import { MdViewHeadline, MdSettings, MdNotifications, MdLanguage, MdHome } from 'react-icons/md'
//import  './styles.css';
import {Headers, List, Element, FieldText, View, Text, Circle} from './styles';

//import api , { config } from '../../Services/api';

const Header = () => (
    <Headers>
        <List>
            <Element>
                <Circle>
                    <IconContext.Provider value={{color: '#fff' , size:'1.5em' }}>
                        <View>
                            <MdViewHeadline/>
                        </View>
                    </IconContext.Provider>
                </Circle>
            </Element>
            <Element>
                <IconContext.Provider value={{color: '#fff' , size:'1.5em' }}>
                    <MdHome/>
                </IconContext.Provider>
            </Element>
            <Element>
                <FieldText>
                    <Text type='text'  placeholder='Pesquisar' id='seach'/>
                </FieldText>
            </Element>
            <Element>
                <IconContext.Provider value={{color: '#fff' , size:'1.5em' }}>
                    <MdLanguage/>
                </IconContext.Provider>
            </Element>
            <Element>
                <IconContext.Provider value={{color: '#fff' , size:'1.5em' }}>
                    <MdNotifications/>
                </IconContext.Provider>
            </Element>
            <Element>
                <IconContext.Provider value={{color: '#fff', size:'1.5em' }}>
                    <MdSettings/>
                </IconContext.Provider>
            </Element>
        </List>
    </Headers>
);

export default Header;
