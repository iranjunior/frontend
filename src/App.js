import React from 'react';
import { connect } from 'react-redux';

import Routes  from './Routes';


function App() {
  return (
          <Routes />
  
  );
}

export default connect(state => ({}))(App);
