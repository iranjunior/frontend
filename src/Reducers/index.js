import auth from './auth'
import user from './user'
import specialities from './specialities'
import uiDesign from './uiDesign'
import vacancies from './vacancies'
import routes from './routes'
import maps from './maps'
import { combineReducers } from 'redux'
import { connectRouter } from 'connected-react-router'

import history from '../Routes/history';


export default combineReducers({auth, user, uiDesign, vacancies, specialities, maps, routes ,router: connectRouter(history)});