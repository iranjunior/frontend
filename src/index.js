import React from 'react';
import ReactDOM from 'react-dom';
<<<<<<< HEAD

import App from './App';
import * as serviceWorker from './serviceWorker';
=======
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { Route } from 'react-router'; 
>>>>>>> 63cd01a06a06c7f0333d67b6947d6e08b1e53b99

import App from './App';
import store   from './Store';
import history from './Routes/history';

ReactDOM.render( (
        
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <Route path='/' component={App}/>
        </ConnectedRouter>
    </Provider>
    
    ), document.getElementById('root'));
