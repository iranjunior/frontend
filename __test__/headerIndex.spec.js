import React from 'react';
import { shallow } from 'enzyme';
import store from '../src/Store';
import configureStore from 'redux-mock-store';
import Header from '../src/Components/Header';
import { middlewares } from '../src/Store';
import configureStore from 'redux-mock-store';

const mockStore = configureStore({ middlewares });

initialState = {};

const store = mockStore(initialState);

describe('Testando redux no Header', () => {
  it('deve renderizar corretamente', () => {
    const wrapper = shallow(<Header />, { context: { store } }).dive();

    expect(wrapper).toMatchSnapshot();
  });
});
