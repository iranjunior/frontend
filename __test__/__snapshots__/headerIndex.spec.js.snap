// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando redux no Header deve renderizar corretamente 1`] = `ShallowWrapper {}`;
